xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -361,8 +363,19 @@ viewers.Comments = Class.create({
 
               if (this.restartNeeded) {
                 this.container.update(response.responseText);
+
+                // If a content is found in submittedcomment that means the submission was not valid.
+                // For instance, the captcha was not accepted.
+                var submittedCommentContainer =  $("submittedcomment");
+                var submittedComment = '';
+                if(submittedCommentContainer) {
+                  submittedComment = submittedCommentContainer.textContent;
+                  this.formDisplayed = false;
```

Comment:
Why this flag?

Reply:
By updating this flag, the form is directly redisplayed to the user when the form validation failed (eg, the captch is not valid). I'll add a comment.